### PR TITLE
Switch defaults

### PR DESCRIPTION
**Pull request by velis74**
Pellucide,

You're probably not going to like this very much, but here's what I had to do in order to create the pull request (complete list of changes follows):
1. Convert the project to Android Studio. This results in all files being moved. The actual changes are only the following:
1.1. Switch.java (renamed from MySwitch.java)
1.2. AndroidSwitchDemoActivity.java (only the MySwitch --> Switch rename changes)
1.3. values/attrs.xml (some properties I had to rename because otherwise I'd get redeclaration errors)
1.4. values/styles.xml (again MySwitch\* --> Switch\* rename)
1.5. layout/main.xml (fixed all the renames)
2. In Switch.java, constructor, I replaced Resources.getColor() calls with constants. Basically Resources.getColor() works in my case because my theming engine requires those colors to be defined. That's not the case with "standard" resources. Though if you use AppCompat library, standard colors are defined in it and can be used for this purpouse. Otherwise these colors have to be extracted using ObtainStyledAttributes() calls.
3. Known issues:
3.1. In Android versions prior to lollipop, the default thumb drawable is inset too much for reasons unknown to me. It should be .5sp, but is actually way more and turns out looking too narrow. I have been unable to determine the cause of this. See new InsetDrawable() call in constructor.
3.2. Sorry about multiple revisions, my TortoiseSVN refused to commit changes to base directory because I added ignore directories.
4. New features added:
4.1. Switch now works without any special attribute assignments. Calling the constructor is enough.
4.2. Default look is lollipop-like (but with text)
4.3. Added thumb transition animations

TL;DR:
Converted project to Android Studio, demo project works, added animations and default lollipop-like look.

### CODE REVIEW

**Comment by pellucide** [comment]
I merged most of your changes to master, I just kept the styling and formatting same as the previous version. This way, it will help any existing users to merge.
The project also supports both Android Studio and Eclipse.
I will close this PR after a final check.

**Comment by velis74** [comment]
Interesting. And Android studio let you? I had to rename those precisely because Android Studio was bugging me about them being redeclared. Maybe it's a MySwitch --> Switch thing? I needed that because I hijack the original Android widget with this one :)
Anyway, have fun and thanks for this excellent widget.

**Comment by pellucide** [comment]
Actually, its gradle that lets you do that. Android studio follows the gradle build script which is structured to allow eclipse to co-exist. But this breaks down when multidex is enabled.
